python/ray/_private/runtime_env/conda_utils.py

```diff
@@ -95,35 +91,37 @@ def get_or_create_conda_env(conda_env_path: str,
             "You can also configure Ray to look for a specific "
             f"Conda executable by setting the {RAY_CONDA_HOME} "
             "environment variable to the path of the Conda executable.")
-    _, stdout, _ = exec_cmd([conda_path, "env", "list", "--json"])
-    envs = json.loads(stdout)["envs"]
 
-    create_cmd = None
-    env_name = _get_conda_env_name(conda_env_path)
-    if base_dir:
-        env_name = f"{base_dir}/{env_name}"
-        if env_name not in envs:
-            create_cmd = [
-                conda_path, "env", "create", "--file", conda_env_path,
-                "--prefix", env_name
-            ]
-    else:
-        env_names = [os.path.basename(env) for env in envs]
-        if env_name not in env_names:
-            create_cmd = [
-                conda_path, "env", "create", "-n", env_name, "--file",
-                conda_env_path
-            ]
+    create_cmd = [
+        conda_path, "env", "create", "--file", conda_yaml_file, "--prefix",
+        prefix
+    ]
 
     if create_cmd is not None:
-        logger.info(f"Creating conda environment {env_name}")
+        logger.info(f"Creating conda environment {prefix}")
         exit_code, output = exec_cmd_stream_to_logger(create_cmd, logger)
         if exit_code != 0:
-            shutil.rmtree(env_name)
+            shutil.rmtree(prefix)
             raise RuntimeError(
-                f"Failed to install conda environment:\n{output}")
+                f"Failed to install conda environment {prefix}:\n{output}")
+
+
+def delete_conda_env(prefix: str,
+                     logger: Optional[logging.Logger] = None) -> bool:
+    if logger is None:
+        logger = logging.getLogger(__name__)
+
+    logger.info(f"Deleting conda environment {prefix}")
+
+    conda_path = get_conda_bin_executable("conda")
+    delete_cmd = [conda_path, "remove", "-p", prefix, "--all", "-y"]
+    exit_code, output = exec_cmd_stream_to_logger(delete_cmd, logger)
```

Comment:
out of curiosity, could we also just do `rm -f` the directory? or this there some special cleanup that this does?

Reply:
I took a look at the source https://github.com/conda/conda/blob/master/conda/cli/main_remove.py and yeah, it looks like the main ingredient is rm rf.  It looks like it edits some metadata too though, but not sure how important that is